src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -381,20 +389,67 @@ async def package_python_dist(
         SetupPyChrootRequest(exported_target, py2=interpreter_constraints.includes_python2()),
     )
 
-    # If commands were provided, run setup.py with them; Otherwise just dump chroots.
+    dist_tgt = exported_target.target
+    wheel = dist_tgt.get(WheelField).value
+    sdist = dist_tgt.get(SDistField).value
+    wheel_config_settings = dist_tgt.get(WheelConfigSettingsField).value or FrozenDict()
+    sdist_config_settings = dist_tgt.get(SDistConfigSettingsField).value or FrozenDict()
+
+    # TODO: When the deprecated SetupPyCommandsField is removed, delete everything from here...
     commands = exported_target.target.get(SetupPyCommandsField).value or ()
-    if commands:
-        validate_commands(commands)
+    if commands or wheel or sdist:
+        if commands:
+            # Override with the legacy commands mechanism.
+            validate_commands(commands)
+            # The original commands might have some options scoped to bdist_wheel and others
+            # scoped to sdist, and possibly some unscoped ones preceeding both. E.g.,
+            # ["--unscoped", "bdist_wheel", "--scoped-to-bdist", "sdist", "--scoped-to-sdist"]
+            # We untangle this as best we can.
+            wheel_cmds: list[str] = []
+            sdist_cmds: list[str] = []
+            scoped_cmds = None  # Will point to one of the above, as we enter its scope.
+            for cmd in commands:
+                if cmd == "bdist_wheel":
+                    wheel = True
+                    scoped_cmds = wheel_cmds
+                elif cmd == "sdist":
+                    sdist = True
+                    scoped_cmds = sdist_cmds
+                elif scoped_cmds:
+                    scoped_cmds.append(cmd)
+                else:
+                    # Unscoped, so use in both.
+                    wheel_cmds.append(cmd)
+                    sdist_cmds.append(cmd)
+            wheel_config_settings = FrozenDict({"--global-option": tuple(wheel_cmds)})
+            sdist_config_settings = FrozenDict({"--global-option": tuple(sdist_cmds)})
+        # ... to here, delete the else branch below, and dedent the remaining code appropriately.
```

Comment:
Yeah, I tried but it got annoying because of the deprecated conditional logic.

Reply:
If we go with wheel and sdist being true by default then this logic gets untangled and easy to break out into a function. So that tips the scale towards doing it...